apigee_edge.module

```diff
@@ -515,7 +515,9 @@ function apigee_edge_api_product_access(EntityInterface $entity, $operation, Acc
   $result = AccessResult::allowedIfHasPermission($account, 'bypass api product access control');
   if ($result->isNeutral()) {
     // Attribute may not exists but in that case it means public.
-    $product_visibility = $entity->getAttributeValue('access') ?? 'public';
+    // Access attribute needs to be set to lower case
+    // to match the portal visibilities that is set in the config apigee_edge.api_product_settings.
+    $product_visibility = $entity->getAttributeValue('access') ? strtolower($entity->getAttributeValue('access')) : 'public';
```

Comment:
The management API also sets lowercase visibility values so if values are not stored lowercased by the form on apigee_edge.api_product_settings then that feature needs to be fixed, not the access control implementation that uses the stored values from Apigee.

![image](https://user-images.githubusercontent.com/1755573/135071513-f098469a-a916-4e98-bc97-dfce91bf4cda.png)

Reply:
Probably it is not a good idea to change all access values on existing API products in an update hook. I would rather emphasize in release notes that the configuration UI worked incorrectly before and every project should validate the value in the `access` attribute. Why? Because they may unintentionally "leveraging" this misbehavior on their site by comparing `Public` with `public` or `Private` with `private`, and do not forget, this is an attribute, so literally, the value can be anything...
![image](https://user-images.githubusercontent.com/1755573/135072593-2d7dd931-2837-4796-9793-3a66d5dcd5cd.png)

Reply:
(This also means that adding the suggested workaround can expose information on existing projects that should not be.)

Reply:
@mxr576 the backend team is going to change the case for the access attribute. 

This will be needed till they roll out the fix. We may be in a state where the attribute could be in different case for a little time.

Reply:
There is additional information on the ticket to make this field an ENUM to only accept certain values.